admin/create-theme/theme-templates.php

```diff
@@ -113,78 +161,127 @@ public static function add_templates_to_local( $export_type ) {
 			wp_mkdir_p( get_stylesheet_directory() . DIRECTORY_SEPARATOR . $template_folders['wp_template'] );
 		}
 
+		// If there is no parts folder, create it.
+		if ( ! is_dir( get_stylesheet_directory() . DIRECTORY_SEPARATOR . $template_folders['wp_template_part'] ) ) {
+			wp_mkdir_p( get_stylesheet_directory() . DIRECTORY_SEPARATOR . $template_folders['wp_template_part'] );
+		}
+
+		// If there is no patterns folder, create it.
+		if ( ! is_dir( get_stylesheet_directory() . DIRECTORY_SEPARATOR . 'patterns' ) ) {
+			wp_mkdir_p( get_stylesheet_directory() . DIRECTORY_SEPARATOR . 'patterns' );
+		}
+
 		foreach ( $theme_templates->templates as $template ) {
-			$template_data = Theme_Blocks::make_template_images_local( $template );
-
-			// If there are images in the template, add it as a pattern
-			if ( ! empty( $template_data->media ) ) {
-				// If there is no templates folder, create it.
-				if ( ! is_dir( get_stylesheet_directory() . DIRECTORY_SEPARATOR . 'patterns' ) ) {
-					wp_mkdir_p( get_stylesheet_directory() . DIRECTORY_SEPARATOR . 'patterns' );
-				}
-
-				// If there are external images, add it as a pattern
-				$pattern                 = Theme_Patterns::pattern_from_template( $template_data );
-				$pattern_link_attributes = array(
-					'slug' => $pattern['slug'],
-				);
-				$template_data->content  = Theme_Patterns::create_pattern_link( $pattern_link_attributes );
 
-				// Write the pattern
-				file_put_contents(
-					get_stylesheet_directory() . DIRECTORY_SEPARATOR . 'patterns' . DIRECTORY_SEPARATOR . $template_data->slug . '.php',
-					$pattern['content']
-				);
-			}
+			$template = Theme_Media::make_template_images_local( $template );
+			$template = self::escape_text_in_template( $template );
+			$template = self::eliminate_environment_specific_content( $template );
+			$template = self::paternize_template( $template );
```

Comment:
I believe `Save Changes` feature should not alter the template structure (probably not done right), specially converting the template to a pattern. It changes the editing experience. I think it should keep the content as is while saving, including escaping of text since the data is being saved to the current site.

All the escaping and patternization should happen with all options that take the theme out of the site.

Reply:
I'm not sure that it does change the editing experience.  To the user the content being in a pattern or a template makes the content just as editable. Just as save-able.

And we wouldn't have the opportunity to escape the text, or use included media, if we don't 'save' that content to a pattern.